From 4b8129ed7e451d3c5ad75f3cb2d7a94c8380e01d Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 24 Feb 2021 13:47:59 +0000
Subject: [PATCH 0963/1000] MANTIS 0060043 change: Create new illumination
 status for "intended" errors

add method AppStateListener::conveyorStateChanged
add parameter componentType to PlatformIF::sendPublicEvent

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1621 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/Massai/platform.idl | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 7c4675dd..2cb43acb 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "5.4";
+    const string interface_version = "5.5";
 
     /** Platform interface for the device components. Via this interface the device
         components communicate with the platform manager internally. */
@@ -325,9 +325,19 @@ module PlatformManagement
       APP_ACTIVE                      // an application is active
     };
 
+
+    enum ConveyorState
+    {
+      CON_READY,
+      CON_MANUAL_MODE, 
+      CON_SWITCHED_OFF,
+      CON_ERROR
+    };
+
     interface AppStateListener
     {
       void appStateChanged(in ApplicationsState state);
+      void conveyorStateChanged(in ConveyorState state);
     };
 
     enum DataHandling
@@ -393,7 +403,7 @@ module PlatformManagement
             @param appRefsAlreadySent - A list of application references those have
                           already received the event via their component listeners
             @return RC_OK if no error occurs, RC_ERROR if the event cannot be sent. */
-        code sendPublicEvent(in any event, in list appRefsAlreadySent);
+        code sendPublicEvent(in string componentType, in any event, in list appRefsAlreadySent);
 
         /** Gets the CUSS platform level.
             @param level - The return value
-- 
2.41.0.windows.1

